### DISCUSSION TITLE
Lösungsansätze?

### DISCUSSION THREAD

**Discussion by Efendi5** [neuefische/questions#565] [Web]
Hallo,
gerade arbeite ich an meiner ResultList und komme seit gestern einfach nicht weiter. Ich sehe wahrscheinlich den Wald vor lauter Bäumen nicht.
Ich bräuchte einen Lösungsansatz, aktuell sieht es so aus, sodass nach jeder Runde die Punkte aktualisiert werden. Ich würde gerne hinbekommen, dass die erste erspielte Runde die Punkte beibehält und in Runde 2/3 neu gezählt wird.
Ich hoffe das Problem ist verständlich. Für gute Lösungsansätze wäre ich sehr dankbar da ich gerne weiter kommen würde.

Ich habe meinen aktuellen Stand committed und gepusht. [Dieser ist unter diesem Link erreichbar.](https://github.com/Efendi5/capstone-project) Der Branch heißt ResultList.
Ich habe auch eine aktuelle Version von Vercel zur Verfügung, dort könnt ihr das gerne testen. [Dieser ist unter diesem Link erreichbar.]
(https://capstone-project-git-resultlist-efendi5.vercel.app/)

Vielen Dank für eure Hilfe!

**Comment by actyvystom** [comment]
Hi @Efendi5 ,
to get detailed support for your question, please provide more detail and code snippets if possible. Guessing from your description, sounds like a global state would be a solution to store points per user per round...

**Reply by Efendi5** [reply]
Ich kann schlecht den ganzen code als snippet hier rein posten, leichter ist es in den verlinken Branch zu gehen und kurz drüber zu schauen, da dort nicht viel code vorhanden ist. Ich bräuchte schon einen Lösungsansatz sonst komm ich nicht weiter...
naja trotzdem danke

**Reply by PReichetanz** [reply]
Hi @Efendi5,
I do not really understand what this means: "Ich würde gerne hinbekommen, dass die erste erspielte Runde die Punkte beibehält und in Runde 2/3 neu gezählt wird."
What exactly do you want to display when? 

If you want to count the points for all rounds each, you need to adapt the `points1` and `points2` state: by now, it's simply a number, but you want to store the points for each round separately. This means that, as soon as the second round starts and the user gives an answer, the `points1` state is overwritten and the points of the previous round are lost.

You could use an object of the following form (or similar):

```jsx
const [points1, setPoints1] = useState({
roundOne: 0,
roundTwo: 0,
roundThree: 0
});
```
You then need to check whether the user clicks the `NextButton` the third time after the third question: if so, in the following round, the `points1` state needs to be updated at the key `roundTwo`.

Hope this helps!
